docs/tutorial.md

```diff
@@ -0,0 +1,183 @@
+
+```
+-IMG ?= controller:latest
++IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
+```
+
+Once done, you do not have to set IMG or any other image variable in the CLI. The following command will build and push an operator image tagged as example.com/memcached-operator:v0.0.1 to Docker Hub:
+
+```
+make docker-build docker-push
+```
+
+### Generating CRD manifests 
+
+```sh
+$ kustomize build config/default > manifest/manifest.yaml
+```
+
+Need to update `manifest.yaml` file.
+
+Update image tag from `controller:latest` to `example.com/memcached-operator:v0.0.1`
+
+Then, remove `ClusterRole` with name `memcached-operator-manager-role` and `ClsuerRoleBinding` with name `memcached-operator-manager-rolebinding`.
+
+Now, create the `ScopeTemplate` and add above removed `ClsuterRole` in spec in `manifests/scopeTemplate.yaml`.
+
+```yaml
+apiVersion: operators.io.operator-framework/v1alpha1
+kind: ScopeTemplate
+metadata:
+  name: scopetemplate-sample-manager
+spec:
+  clusterRoles:
+  - generateName: manager-role
+    rules:
+    - apiGroups:
+      - "*"
+      resources:
+      - "*"
+      verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+    - apiGroups:
+      - cache.example.com
+      resources:
+      - memcacheds/finalizers
+      verbs:
+      - update
+    - apiGroups:
+      - cache.example.com
+      resources:
+      - memcacheds/status
+      verbs:
+      - get
+      - patch
+      - update
+    subjects:
+    - kind: ServiceAccount
+      name: memcached-operator-controller-manager
+      namespace: memcached-operator-system
+```
+
+Before this install `ScopeTemplate` CRD from `oria-operator`.
+
+```
+$ make install
+```
+
+Then, apply manifest and look for the pod logs.
+
+```
+$ kubectl apply -f manifest
+namespace/memcached-operator-system created
+customresourcedefinition.apiextensions.k8s.io/memcacheds.cache.example.com created
+serviceaccount/memcached-operator-controller-manager created
+role.rbac.authorization.k8s.io/memcached-operator-leader-election-role created
+...
+```
+
+The `memcached-operator` is running successfully with some errors. Check the pod logs and now you will be able to see that pod is complaining about `RBAC`.
+
+```
+$ kubectl get pods -n memcached-operator-system
+NAME                                                     READY   STATUS    RESTARTS   AGE
+memcached-operator-controller-manager-684554467c-4jflr   2/2     Running   0          9m45s
+
+$ kubectl describe pod memcached-operator-controller-manager-684554467c-4jflr manager -n memcached-operator-system
+```
+
+Now, run the `oria-operator`
+
+```
+$ make run
+```
+
+Create `ScopeInstance` that refers to `ScopeTemplate` and apply it.
+
+```yaml
+apiVersion: operators.io.operator-framework/v1alpha1
+kind: ScopeInstance
+metadata:
+  name: scopeinstance-sample-manager
+spec:
+  scopeTemplateName: scopetemplate-sample-manager
+```
+
+```
+$ kubectl apply -f operators_v1_scopeinstance.yaml
+```
```

Comment:
There is a handy shortcut for tutorials/demos that I like to use when mentioning that a smaller manifest should be applied. Instead of having them create the YAML file and apply it by hand I normally provide a command that will apply it for them. In this case I would set the command as:
```
cat << EOF | kubectl apply -f -
apiVersion: operators.io.operator-framework/v1alpha1
kind: ScopeInstance
metadata:
  name: scopeinstance-sample-manager
spec:
  scopeTemplateName: scopetemplate-sample-manager
EOF
```
This way they can copy and paste the command and follow along easier than having to create a bunch of files.

We don't have to do this in this tutorial, but I just figured I would share since it is how I normally create mine :)

Reply:
updated above content as like above.